src/tests/system/tests/test_ipa.py

```diff
@@ -126,6 +129,78 @@ def test_ipa__hostpublickeys_by_ip(client: Client, ipa: IPA, public_keys: list[s
         assert f"{ip} {key}" in result.stdout_lines, "Did not get expected public keys!"
 
 
+@pytest.mark.ticket(gh=7583)
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hostpublickeys_by_name_with_port(client: Client, ipa: IPA, public_keys: list[str]):
+    """
+    :title: sss_ssh_knownhosts returns public keys by host name with port
+    :setup:
+        1. Create host with SSH key
+        2. Configure SSSD with SSH responder
+        3. Start SSSD
+    :steps:
+        1. Lookup SSH key
+    :expectedresults:
+        1. All public keys were printed
+    :customerscenario: False
+    """
+    hostname = f"ssh-host.{ipa.domain}"
+    ip = "10.255.251.10"
+
+    ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+    port = 3333
```

Comment:
It would be cleaner to set the port number together with the hostname and IP address. I mean, in line 150.

Reply:
If we add with `hostname` then `ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)` is failing as we don't support that format in IPA.

Reply:
What I mean is:
```
   hostname = f"ssh-host.{ipa.domain}"
   ip = "10.255.251.10"
   port = 3333

   ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
```

Reply:
updated.